### ISSUE TITLE
convert "csvPaser" file into ES6

### ISSUE THREAD

**Issue by atherdon** [GroceriStar/food-datasets-csv-parser#13]
i think this task is pretty similar to work, that we're doing at generator repository. I think it can be done by our juniors.

other files are not important - because soon we'll get rid of them when this module will work well as separated entity

**Comment by atherdon** [comment]
i think this can be the next task for @svr8

**Comment by atherdon** [comment]
While Vadim reviewing our progress at generator project - @AndrewSerra you can jump into this task as well. But before we'll do an actuall work - let's discuss a plan how we'll do it

**Comment by atherdon** [comment]
Andrew, you can jump into this task. a generator will also be used inside of this module, so we'll be able to test it a few places in the same time

**Comment by atherdon** [comment]
and as right now it partially worked with this es5 module - i hope it will work also with es6 version as well

**Comment by AndrewSerra** [comment]
okay, I'll start this. @svr8 did you do anything yet?

**Comment by AndrewSerra** [comment]
are we changing all csv parsing related files in this task?

**Comment by svr8** [comment]
@AndreSerra @atherdon Nope, I haven't done much on this yet. I was looking into testing and related issues. I'll start on this issue after I've fixed the tests for fileSystem.js. I hope it is alright.

**Comment by atherdon** [comment]
let's list files that we have at src folder(not including "projects" for
now).
So we'll be able to discuss each file separately.

I think it should be pretty easy task, because we did similar work at
generator module

>

**Comment by AndrewSerra** [comment]
here is the src directory structure (without the mutliple .json files. to discuss, I only left one if there are many that makes the listing too long), 
TLDR -> removed most json files under USFA/Nutrition and USFA/Product for this visualization:

```
.
├── CCCSVParser.js
├── FoodComposition
│   ├── FoodComposition\ -\ Finland.json
│   ├── FoodComposition\ -\ France.json
│   ├── FoodComposition\ -\ Germany.json
│   ├── FoodComposition\ -\ Italy.json
│   ├── FoodComposition\ -\ Netherlands.json
│   ├── FoodComposition\ -\ Sweden.json
│   ├── FoodComposition\ -\ United\ Kingdom.json
│   ├── FoodComposition.json
│   ├── csv_parser.js
│   └── files.js
├── USFA
│   ├── Derivation_Code_Description
│   │   ├── Derivation_Code_Description1.json
│   │   └── parser.js
│   ├── Nutrition
│   │   ├── Nutrient01.json
│   │   ├── files.js
│   │   └── parser.js
│   ├── Product
│   │   ├── Products01.json
│   │   └── parser.js
│   ├── Readme.md
│   ├── Serving_Size
│   │   ├── Serving_Size1.json
│   │   └── parser.js
│   └── files.js
├── fileSystem.js
├── index.js
├── utils.js
└── writeFile.js
```

**Comment by atherdon** [comment]
```
├── CCCSVParser.js
├── fileSystem.js
├── index.js
├── utils.js
└── writeFile.js
```

are our first candidates for es6 transition

**Comment by atherdon** [comment]
add your cool structure at readme file please

**Comment by AndrewSerra** [comment]
Sure, also if anyone uses a mac, you could use `brew install tree` pretty cool tool. I just found it and used it to generate that.

**Comment by AndrewSerra** [comment]
> ```
> ├── CCCSVParser.js
> ├── fileSystem.js
> ├── index.js
> ├── utils.js
> └── writeFile.js
> ```
> 
> are our first candidates for es6 transition

I can handle those. Do we have the same scripts used in food-static-files-generator to not get errors?

**Comment by AndrewSerra** [comment]
Another thing I have found if it can be useful for anyone here is a [source](https://engineering.carsguide.com.au/es5-vs-es6-syntax-6c8350fa6998) about differences between ES5 and ES6.

**Comment by atherdon** [comment]
let's start with cccsvparser first.
but let's do it in different way.
can you plit it into a few files, so it became less long and easy to digest?

as you may find - utils and writeFile is the same files that we have at generator.
for sure, we can just replace of delete them. but it's a brutal way and it can cause some issues.
first of issue that right now ccsv parser, even with es5 is working. so i want to go slowly in order to don't 

---

you can add structure and that link to es5-6 differences at readme as well

---

i have more advanced notes at this repo, but will share them lately.

we also have 1 dataset that will require to create a custom parser from scratch and foodcomposition require an upgrade, but this is in the future

**Comment by AndrewSerra** [comment]
Okay, I can check cccsvparser in the morning and figure something out to split it up

**Comment by AndrewSerra** [comment]
I'm also handling some todos, I made a join path function in utils so the code gets less complicated to read

**Comment by AndrewSerra** [comment]
@atherdon I have made some changes and added a function so you can see and if you like it we can start using that as well.

**Comment by atherdon** [comment]
Ok, what we should do next @AndrewSerra ?

**Comment by AndrewSerra** [comment]
we could convert utils.js and writeFile.js to ES6 style They are not converted yet

**Comment by AndrewSerra** [comment]
Also, do you think we can use the function I wrote? If yes, we could clean up code by replacing the current path.join() and path.resolve() to the one I wrote in utils.js

**Comment by atherdon** [comment]
can you give me a codeline to your functions?

**Comment by AndrewSerra** [comment]
Here it is (located in src/utils.js):
```
/**
 * For joinPath()
 * @param {Arrat} arr
 * @param {Bool} resolve
 */
const joinPath = (arr, useResolve=false) => {

  if(useResolve) {
    return resolve(...arr);
  }

  return join(...arr);
}
```

**Comment by atherdon** [comment]
1) let's move this method outside from utils.js first. because i assume soon we'll move to use methods that came from util.js at generator method

2) utils.js and writeFile.js to ES6 style 
Can you tell me - do we have a difference inside of these files with generator files?
or maybe what methods we're using from both of this files, that are different
https://github.com/GroceriStar/food-static-files-generator/blob/master/src/utils.js
https://github.com/GroceriStar/food-static-files-generator/blob/master/src/writeFile.js

**Comment by atherdon** [comment]
what do you also think if we'll move methods from cccsvparser and move them into index.js?
i think as it's a separated module - we can keep those methods at the main file

**Comment by AndrewSerra** [comment]
@atherdon I think I answered this in the [issue](https://github.com/GroceriStar/food-datasets-csv-parser/issues/33#issuecomment-511412101).

**Comment by AndrewSerra** [comment]
What file are we continuing with this issue? @svr8 did you do anything else?

**Comment by atherdon** [comment]
we need to decide what to do next here

>

**Comment by atherdon** [comment]
can you check and tell me which files from utisl.js are actually used?

i think we need to delete all methods that we don't using and convert
utils. to es6

>

**Comment by AndrewSerra** [comment]
should I convert it while checking?

**Comment by AndrewSerra** [comment]
`checkFilePath()` ,`getFileInfo()` are not used. I'll remove them now and change to ES6 style.

**Comment by atherdon** [comment]
cool

**Comment by AndrewSerra** [comment]
utils.js is done and merged. whats next for this issue

**Comment by atherdon** [comment]
moving to separated issue
